### PR TITLE
[#115322155] Use correct version of fly for Bootstrap/Deployer Concourse

### PR DESCRIPTION
**Pull request by alext**
## What

We should use the correct version of fly for each Concourse. Assuming that the Bootstrap Concourse and Deployer Concourse will be using different versions, either now or again in the future.

As part of this we've taken the opportunity to refactor how the environment is setup for these scripts. See commit messages for more details.

## How to review

Use this branch to push pipelines to a bootstrap concourse (`./vagrant/deploy.sh`). Push pipelines to the deployer concourse (`make dev`). This should create and use 2 different fly commands in the `bin` directory. This should not touch any `fly` binaries that happen to be on your path.

Check that the `bosh-cli` script works correctly.

## Who can review

Anyone but @timmow or myself.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -61,5 +61,4 @@ disable_auto_delete:
 	$(eval export DISABLE_AUTODELETE=1)
 
 deploy_pipelines:
-	concourse/scripts/pipelines-bosh-cloudfoundry.sh
-	concourse/scripts/pipelines-failure-testing.sh
+	concourse/scripts/push-deployer-pipelines.sh
```
Comment (saliceti) [inline Makefile]: I'd rather avoid creating another wrapper script if possible. I've made a simple attempt in branch fly_versions_2, isn't it good enough?

File: concourse/scripts/fly_sync_and_login.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  echo "Downloading fly command..."
+  curl "$FLY_CMD_URL" -L -f -k -o "$FLY_CMD" -u "${CONCOURSE_ATC_USER}:${CONCOURSE_ATC_PASSWORD}"
+  chmod +x "$FLY_CMD"
+fi
+
+echo "Doing fly login & sync"
+echo -e "${CONCOURSE_ATC_USER}\n${CONCOURSE_ATC_PASSWORD}" | \
+  $FLY_CMD -t "${FLY_TARGET}" login -k --concourse-url "${CONCOURSE_URL}"
+
+$FLY_CMD -t "${FLY_TARGET}" sync
```
Comment (saliceti) [inline concourse/scripts/fly_sync_and_login.sh]: This always downloads the fly binary. Can we add an option like FLY_SYNC=false to disable it when we're developing?

Comment (alext) [inline concourse/scripts/fly_sync_and_login.sh]: Good call. Added.

Comment (dcarley) [inline concourse/scripts/fly_sync_and_login.sh]: If everyone starts using this habitually, are we going to be bitten when we next upgrade Concourse?

How about checking the modified time on the file and only updating if it's N hours/days old?

Comment (alext) [inline concourse/scripts/fly_sync_and_login.sh]: > How about checking the modified time on the file and only updating if it's N hours/days old?

I like it.  I'll look at doing that.  What do you think is a reasonable value for `N`?

Comment (dcarley) [inline concourse/scripts/fly_sync_and_login.sh]: A day?

Comment (alext) [inline concourse/scripts/fly_sync_and_login.sh]: I've now implemented this, and rebased.

File: Makefile (hunk 1 above)
Comment (alext) [inline Makefile]: This has now been refactored so that is has a single environment script.

**Comment by alext** [comment]
I think I've now addressed all the comments here. Ready to be re-tested.

**Comment by saliceti** [comment]
Tested OK
Thanks! 👍

**Pull request merged**